src/bsoncxx/types.hpp

```diff
@@ -122,10 +158,14 @@ struct BSONCXX_API b_utf8 {
     template <typename T,
               typename std::enable_if<!std::is_same<b_utf8, typename std::decay<T>::type>::value,
                                       int>::type = 0>
-    BSONCXX_INLINE explicit b_utf8(T&& t) : value(std::forward<T>(t)) {}
+    BSONCXX_INLINE BSONCXX_DEPRECATED explicit b_utf8(T&& t) : value(std::forward<T>(t)) {}
 
     stdx::string_view value;
 
+    operator b_string() {
+        return b_string{value};
+    }
```

Comment:
You'll notice this can't be round-tripped, i.e., `operator b_utf8` isn't defined in `b_string`. This is because we would have to create a `b_utf8` in the definition and the constructor for the `b_utf8` is deprecated. Again, this is a public header so including the private header containing `BSONCXX_SUPPRESS_DEPRECATION_WARNINGS...` would be a bad idea. I _could_ create a public version of the suppression warnings but that would just allow users to bypass the deprecation warning, assuming that even works.

Reply:
I agree that we should not create a public version of the suppression warnings.

With the typedef strategy I mentioned in a different comment, I think this issue would go away.

Reply:
Done